dandiapi/api/services/publish/__init__.py

```diff
@@ -106,11 +106,15 @@ def _publish_dandiset(dandiset_id: int) -> None:
 
     Calling `_lock_dandiset_for_publishing()` is a precondition for calling this function.
     """
-    old_version: Version = (
-        Dandiset.objects.get(id=dandiset_id).versions.select_for_update().get(version='draft')
-    )
-
     with transaction.atomic():
+        old_version: Version = (
+            Version.objects.all()
```

Comment:
I don't believe the `.all()` is needed, correct?

Reply:
Fixed.